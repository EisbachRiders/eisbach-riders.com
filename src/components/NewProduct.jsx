import React, { useState } from 'react'
import PropTypes from 'prop-types'
import Img from 'gatsby-image'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/styles'
import Typography from '@material-ui/core/Typography'
import Paper from '@material-ui/core/Paper'
import Hidden from '@material-ui/core/Hidden'
import ProductDialog from './ProductDialog'
import { newProduct } from '../data/ProductData'

const useStyles = makeStyles(theme => ({
  root: {
    paddingTop: 30,
    paddingBottom: 30,
    paddingLeft: 15,
    paddingRight: 15,
    background: `linear-gradient(to bottom, ${
      theme.status.greyBlue
    } 50%, #fff 50%)`,
    display: 'flex',
    alignItems: 'center',
    [theme.breakpoints.up('sm')]: {
      paddingLeft: 60,
      paddingRight: 60,
    },
    [theme.breakpoints.up('md')]: {
      paddingTop: 45,
      paddingBottom: 45,
      paddingLeft: 170,
      paddingRight: 170,
    },
  },
  img: {
    height: 'auto',
    width: '60%',
    [theme.breakpoints.up('md')]: {
      marginLeft: 50,
    },
  },
  paper: {
    padding: 30,
    zIndex: 2,
    display: 'flex',
    flexDirection: 'column',
    width: '90%',
    alignItems: 'center',
    textAlign: 'center',
    [theme.breakpoints.up('sm')]: {
      flexBasis: '50%',
      marginLeft: -100,
    },
    [theme.breakpoints.up('md')]: {
      width: 400,
      margin: -100,
    },
  },
  logo: {
    width: 125,
  },
  title: {
    textTransform: 'uppercase',
    fontWeight: 700,
    letterSpacing: 3,
    paddingBottom: 5,
    fontSize: 16,
  },
}))

function NewProduct({ img, detailImg }) {
  const classes = useStyles()
  const { t } = useTranslation()
  const [isDialogOpen, setDialog] = useState(false)

  return (
    <div className={classes.root} id="products">
      <Hidden xsDown>
        <Img
          fluid={img.node.fluid}
          alt={t('newProduct.imgAlt')}
          className={classes.img}
        />
      </Hidden>
      <Paper className={classes.paper} elevation={10}>
        <Img
          fluid={detailImg.node.fluid}
          alt={t('newProduct.logo')}
          className={classes.logo}
        />
        <Typography className={classes.title}>Bee Swell Surf Wax</Typography>
        <Typography>{t('newProduct.text')}</Typography>
        <ProductDialog
          product={newProduct[0]}
          buttonText={t('newProduct.more')}
          handleDialogOpen={() => setDialog(true)}
          handleDialogClose={() => setDialog(false)}
          isDialogOpen={isDialogOpen}
        />
      </Paper>
    </div>
  )
}

NewProduct.propTypes = {
  img: PropTypes.object.isRequired,
  detailImg: PropTypes.object.isRequired,
}

export default NewProduct
