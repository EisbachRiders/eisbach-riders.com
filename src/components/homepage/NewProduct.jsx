import React, { useState } from 'react'
import SwipeableViews from 'react-swipeable-views'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/core/styles'
import clsx from 'clsx'
import Typography from '@material-ui/core/Typography'
import Paper from '@material-ui/core/Paper'
import Hidden from '@material-ui/core/Hidden'
import Button from '@material-ui/core/Button'
import IconButton from '@material-ui/core/IconButton'
import MobileStepper from '@material-ui/core/MobileStepper'
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft'
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight'
import img_wax from '../../assets/images/newProduct/surfWax.jpg'
import img_hangers from '../../assets/images/newProduct/hangers.png'
import logo_wax from '../../assets/images/newProduct/waxLogo.jpg'
import logo_hangers from '../../assets/images/newProduct/hangerLogo.png'
import Container from '../ui/Container'

const useStyles = makeStyles(theme => ({
  container: {
    background: `linear-gradient(to bottom, ${theme.status.greyBlue} 50%, #fff 50%)`,
    flexDirection: 'column',
  },
  swipeable: {
    flexShrink: 'none',
  },
  textContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  img: {
    height: '60%',
    width: '60%',
    [theme.breakpoints.up('lg')]: {
      marginLeft: 50,
    },
  },
  paper: {
    padding: 15,
    textAlign: 'center',
    boxShadow: 'none',
    [theme.breakpoints.up('sm')]: {
      flexBasis: '50%',
      marginLeft: -100,
      marginRight: 10,
      marginBottom: 10,
      boxShadow:
        '0px 6px 6px -3px rgba(0,0,0,0.2), 0px 10px 14px 1px rgba(0,0,0,0.14), 0px 4px 18px 3px rgba(0,0,0,0.12)',
    },
    [theme.breakpoints.up('md')]: {
      width: 400,
      marginLeft: -100,
      marginBottom: 0,
    },
  },
  logo: {
    width: 125,
  },
  title: {
    textTransform: 'uppercase',
    fontWeight: 700,
    letterSpacing: 3,
    paddingBottom: 5,
    fontSize: 16,
  },
  mobileStepper: {
    background: 'transparent',
  },
  button: {
    marginTop: 30,
  },
  nextButton: {
    [theme.breakpoints.up('sm')]: {
      marginTop: -430,
      marginRight: -70,
      zIndex: 200,
    },
  },
  backButton: {
    [theme.breakpoints.up('sm')]: {
      marginTop: -430,
      marginLeft: -40,
      zIndex: 200,
    },
  },
}))

function NewProduct() {
  const classes = useStyles()
  const { t } = useTranslation()
  const [activeStep, setActiveStep] = useState(0)

  const handleIndexChange = activeStep => {
    setActiveStep(activeStep)
  }

  const handleNext = () => {
    setActiveStep(activeStep + 1)
  }

  const handleBack = () => {
    setActiveStep(activeStep - 1)
  }

  const products = [
    {
      product: 'hangers',
      title: 'Hidden Surfboard Wall Rack',
      amazon: 'https://shop.eisbach-riders.com/product/surfboard-wall-rack/',
      img: img_hangers,
      logo: logo_hangers,
    },
    {
      product: 'wax',
      title: 'Bee Swell Surf Wax',
      amazon: 'https://shop.eisbach-riders.com/shop/bee-swell-surf-wax/',
      img: img_wax,
      logo: logo_wax,
    },
  ]
  return (
    <Container className={classes.container}>
      <SwipeableViews
        enableMouseEvents
        index={activeStep}
        onChangeIndex={handleIndexChange}
        className={classes.swipeable}
      >
        {products.map(elem => (
          <div
            key={`newproduct${elem.product}`}
            className={classes.textContainer}
          >
            <Hidden xsDown>
              <img
                src={elem.img}
                alt={t('newProduct.imgAlt')}
                className={classes.img}
              />
            </Hidden>
            <Paper className={classes.paper} elevation={10}>
              <img
                src={elem.logo}
                alt={t(elem.product)}
                className={classes.logo}
              />
              <Typography className={classes.title}>{elem.title}</Typography>
              <Typography>{t(`newProduct.text_${elem.product}`)}</Typography>
              <Button
                color="primary"
                variant="outlined"
                className={clsx(classes.button)}
                href={elem.amazon}
              >
                {t('common.more')}
              </Button>
            </Paper>
          </div>
        ))}
      </SwipeableViews>
      <MobileStepper
        steps={2}
        position="static"
        activeStep={activeStep}
        className={classes.mobileStepper}
        nextButton={
          <IconButton
            size="small"
            className={classes.nextButton}
            aria-label={t('common.scrollLeft')}
            onClick={handleNext}
            disabled={activeStep === 1}
          >
            <KeyboardArrowRight />
          </IconButton>
        }
        backButton={
          <IconButton
            size="small"
            className={classes.backButton}
            aria-label={t('common.scrollRight')}
            onClick={handleBack}
            disabled={activeStep === 0}
          >
            <KeyboardArrowLeft />
          </IconButton>
        }
      />
    </Container>
  )
}

export default NewProduct
