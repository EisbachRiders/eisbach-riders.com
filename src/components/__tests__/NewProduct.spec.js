import React from 'react'
import { render } from 'react-testing-library'
import NewProduct from '../NewProduct'

describe('NewProduct', () => {
  it('renders correctly', () => {
    const mockProps = {
      detailImg: {},
      img: {},
    }
    const component = render(<NewProduct {...mockProps} />)
    expect(component).toMatchSnapshot()
  })
})
