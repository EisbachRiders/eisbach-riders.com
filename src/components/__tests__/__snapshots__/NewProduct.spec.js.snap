// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProduct renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hook-root-25f6vz"
        id="products"
      >
        <div
          class="Hook-img-2qqtdd gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            style="width: 100%;"
          />
          <picture>
            <source
              sizes="mockString"
              srcset="mockString"
            />
            <img
              alt="surfer"
              src="mockString"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcSet="mockString" sizes="mockString" /&gt;&lt;img src="mockString" alt="surfer" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <div
          class="MuiPaper-root-1 MuiPaper-elevation10-13 MuiPaper-rounded-2 Hook-paper-njyc45"
        >
          <div
            class="Hook-logo-9m77sf gatsby-image-wrapper"
            style="position: relative; overflow: hidden;"
          >
            <div
              style="width: 100%;"
            />
            <picture>
              <source
                sizes="mockString"
                srcset="mockString"
              />
              <img
                alt="logo"
                src="mockString"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;source srcSet="mockString" sizes="mockString" /&gt;&lt;img src="mockString" alt="logo" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
          <p
            class="MuiTypography-root-28 MuiTypography-body1-37 Hook-title-el340b"
          >
            Bee Swell Surf Wax
          </p>
          <p
            class="MuiTypography-root-28 MuiTypography-body1-37"
          >
            We are so excited to be able to bring you this premium, eco-friendly surf wax! Ride waves with confidence and a clear conscience! Bee Swell Eco-Friendly surf wax is made with 100% pure beeswax that has been ethically sourced by natural beekeepers. Proceeds help to provide pollinator habitat and resources for honeybee rescues.
          </p>
          <button
            class="MuiButtonBase-root-81 MuiButton-root-64 MuiButton-outlined-69 MuiButton-outlinedSecondary-71 Hook-button-1ihz2ti"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-65"
            >
              more info
            </span>
            <span
              class="MuiTouchRipple-root-101"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hook-root-25f6vz"
      id="products"
    >
      <div
        class="Hook-img-2qqtdd gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          style="width: 100%;"
        />
        <picture>
          <source
            sizes="mockString"
            srcset="mockString"
          />
          <img
            alt="surfer"
            src="mockString"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcSet="mockString" sizes="mockString" /&gt;&lt;img src="mockString" alt="surfer" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <div
        class="MuiPaper-root-1 MuiPaper-elevation10-13 MuiPaper-rounded-2 Hook-paper-njyc45"
      >
        <div
          class="Hook-logo-9m77sf gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            style="width: 100%;"
          />
          <picture>
            <source
              sizes="mockString"
              srcset="mockString"
            />
            <img
              alt="logo"
              src="mockString"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcSet="mockString" sizes="mockString" /&gt;&lt;img src="mockString" alt="logo" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <p
          class="MuiTypography-root-28 MuiTypography-body1-37 Hook-title-el340b"
        >
          Bee Swell Surf Wax
        </p>
        <p
          class="MuiTypography-root-28 MuiTypography-body1-37"
        >
          We are so excited to be able to bring you this premium, eco-friendly surf wax! Ride waves with confidence and a clear conscience! Bee Swell Eco-Friendly surf wax is made with 100% pure beeswax that has been ethically sourced by natural beekeepers. Proceeds help to provide pollinator habitat and resources for honeybee rescues.
        </p>
        <button
          class="MuiButtonBase-root-81 MuiButton-root-64 MuiButton-outlined-69 MuiButton-outlinedSecondary-71 Hook-button-1ihz2ti"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-65"
          >
            more info
          </span>
          <span
            class="MuiTouchRipple-root-101"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
